presto-native-execution/src/test/java/com/facebook/presto/nativeworker/ContainerQueryRunner.java

```diff
@@ -0,0 +1,304 @@
+import com.facebook.presto.sql.planner.ConnectorPlanOptimizerManager;
+import com.facebook.presto.sql.planner.NodePartitioningManager;
+import com.facebook.presto.testing.MaterializedResult;
+import com.facebook.presto.testing.QueryRunner;
+import com.facebook.presto.testing.TestingAccessControlManager;
+import com.facebook.presto.transaction.TransactionManager;
+import org.testcontainers.containers.BindMode;
+import org.testcontainers.containers.Container;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.Network;
+import org.testcontainers.containers.wait.strategy.Wait;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.locks.Lock;
+import java.util.logging.Logger;
+
+import static com.facebook.presto.testing.TestingSession.testSessionBuilder;
+import static org.testng.Assert.fail;
+
+public class ContainerQueryRunner
+        implements QueryRunner
+{
+    private static final Network network = Network.newNetwork();
+    private static final String PRESTO_COORDINATOR_IMAGE = System.getProperty("coordinatorImage", "presto-coordinator:latest");
+    private static final String PRESTO_WORKER_IMAGE = System.getProperty("workerImage", "presto-worker:latest");
+    private static final String CONTAINER_TIMEOUT = System.getProperty("containerTimeout", "120");
+    private static final String CLUSTER_SHUTDOWN_TIMEOUT = System.getProperty("clusterShutDownTimeout", "10");
+    private static final String BASE_DIR = System.getProperty("user.dir");
+    private static final int DEFAULT_COORDINATOR_PORT = 8080;
+    private static final String TPCH_CATALOG = "tpch";
+    private static final String TINY_SCHEMA = "tiny";
+    private static final int DEFAULT_NUMBER_OF_WORKERS = 4;
+    private final GenericContainer<?> coordinator;
+    private final List<GenericContainer<?>> workers = new ArrayList<>();
+    private final int coordinatorPort;
+    private final String catalog;
+    private final String schema;
+    private final int numberOfWorkers;
+    Logger logger = Logger.getLogger(ContainerQueryRunner.class.getName());
+
+    public ContainerQueryRunner()
+            throws InterruptedException, IOException
+    {
+        this(DEFAULT_COORDINATOR_PORT, TPCH_CATALOG, TINY_SCHEMA, DEFAULT_NUMBER_OF_WORKERS);
+    }
+
+    public ContainerQueryRunner(int coordinatorPort, String catalog, String schema, int numberOfWorkers)
+            throws InterruptedException, IOException
+    {
+        this.coordinatorPort = coordinatorPort;
+        this.catalog = catalog;
+        this.schema = schema;
+        this.numberOfWorkers = numberOfWorkers;
+
+        // TODO: This framework is tested only in Ubuntu x86_64, as there is no support to run the native docker images in ARM based system,
```

Comment:
Can you create an issue for this once it's merged?

Reply:
The work is in progress https://github.com/prestodb/presto/pull/23275